tests/WE2E/setup_WE2E_tests.sh

```diff
@@ -1,5 +1,10 @@
 #!/usr/bin/env bash
```

Comment:
Could you update the usage comments of this script below? I was able to run fundamental tests witch
```
./setup_WE2E_tests.sh hera zrtrr intel custom
```
But not a custom test file `custom.hera.intel.com`
```
./setup_WE2E_tests.sh hera zrtrr intel custom
```

Reply:
The script should be able to work with any custom file name, suite name, or test name; I confirmed this worked through the `./setup_WE2E_tests.sh` script with my own test on Jet. Can you send me the command that is failing for you and the directory? The above two commands appear to be the same.

Reply:
@mkavulich I think test files like `custom.hera.intel.com` are not recognized because such patterns are only parsed for `fundamental` and `comprehensive` tests according to the logic in line 72 of `run_WE2E_tests.py`
```
$ ./setup_WE2E_tests.sh hera zrtrr intel custom
/home/Daniel.Abdi
hera
zrtrr
intel
custom

Modules based on Lua: Version 8.5.2  2021-05-12 12:44 -05:00
    by Robert McLay mclay@tacc.utexas.edu


Currently Loaded Modules:
  1) rocoto/1.3.3   2) miniconda3/4.12.0   3) wflow_hera

 

Logging set up successfully
Checking if ['custom'] is a valid test name
Checking if ['custom'] is a valid test suite
Checking if ['custom'] is a file containing test names

*********************************************************************
FATAL ERROR:
Experiment generation failed. See the error message(s) printed below.
For more detailed information, check the log file from the workflow
generation script: log.run_WE2E_tests
*********************************************************************

Traceback (most recent call last):
  File "/scratch2/BMC/gsd-hpcs/Daniel.Abdi/ufs-srweather-app/tests/WE2E/./run_WE2E_tests.py", line 534, in <module>
    run_we2e_tests(homedir,args)
  File "/scratch2/BMC/gsd-hpcs/Daniel.Abdi/ufs-srweather-app/tests/WE2E/./run_WE2E_tests.py", line 109, in run_we2e_tests
    raise FileNotFoundError(dedent(f"""
FileNotFoundError: 
The specified 'tests' argument '['custom']'
does not appear to be a valid test name, a valid test suite, or a file
containing valid test names.

Check your inputs and try again.
```

Reply:
@danielabdi-noaa Sorry, I still don't understand what behavior you are expecting. It appears you gave the script `custom` as an argument; if that's not a file name then I'm not aware of what functionality that's supposed to have.

Reply:
The file I have is `custom.hera.intel.com` not `custom`. With the new `setup_WE2E_tests.sh`, none of the below two calls work
```
./setup_WE2E_tests.sh hera zrtrr intel custom.hera.intel.com
./setup_WE2E_tests.sh hera zrtrr intel custom
```
while with the old one, they both do. Line 72 of `run_WE2E_tests.py` is a new addition that was absent in the old one, and I believe is the cause of this problem. I can understand if the behaviour is changed now, but that should be documented if so.

Reply:
@mkavulich Ok sounds good. In that case the usage of the script should be modified accordingly
```
echo "       test_type     [optional] test type: fundamental or comprehensive or all or any other name"
```
The `test_type` can not be "any other name" now, and also due to the consoldation of the three arguments maybe add also that you can pass a specific test case name or a path to a file containing list of tests.

Reply:
Thanks Daniel, I have updated the usage statement in `./setup_WE2E_tests.sh` to accurately reflect the new argument behavior